backend/corpora/gallica/gallica.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime
+
+from bs4 import BeautifulSoup
+from ianalyzer_readers.xml_tag import Tag
+from ianalyzer_readers.extract import Metadata, XML
+import requests
+
+from addcorpus.python_corpora.corpus import XMLCorpusDefinition
+from addcorpus.python_corpora.corpus import FieldDefinition
+from addcorpus.python_corpora.filters import DateFilter
+from addcorpus.es_mappings import (
+    keyword_mapping,
+    date_mapping,
+    main_content_mapping,
+)
+
+
+def get_content(content: BeautifulSoup) -> str:
+    """Return text content in the parsed HTML file from the `texteBrut` request
+    This is contained in the first <p> element after the first <hr> element.
+    """
+    text_nodes = content.find("hr").find_next_siblings("p")
+    return "".join([node.get_text() for node in text_nodes])
+
+
+class Gallica(XMLCorpusDefinition):
+
+    languages = ["fr"]
+    data_url = "https://gallica.bnf.fr"
+    corpus_ark = ""
+
+    def sources(self, start: datetime, end: datetime):
+        # obtain list of ark numbers
+        response = requests.get(
+            f"{self.data_url}/services/Issues?ark=ark:/12148/{self.corpus_ark}/date"
+        )
+        year_soup = BeautifulSoup(response.content, "xml")
+        years = [
+            year.string
+            for year in year_soup.find_all("year")
+            if int(year.string) >= start.year and int(year.string) <= end.year
+        ]
+        for year in years:
+            response = requests.get(
```

Comment:
This is half question half comment:

What happens in case of server being unresponsive, internet connection failing for a split second, stuff like that?
In the ideal case harvesting and indexing would be split operations so one failing does not lead to starting over completely. Unsure ho wwe approached this in previous API-exposed corpora, hence the question.

Reply:
Discussed this a bit with Luka, and this is how we usually approach online corpora. So ignore the comment!